src/Server.php

```diff
@@ -523,20 +543,14 @@ public function makeImage($path, array $params)
         }
 
         try {
-            $write = $this->cache->write(
+            $this->cache->write(
                 $cachedPath,
                 $this->api->run($tmp, $this->getAllParams($params))
             );
-
-            if ($write === false) {
-                throw new FilesystemException(
-                    'Could not write the image `'.$cachedPath.'`.'
-                );
-            }
-        } catch (FileExistsException $exception) {
-            // This edge case occurs when the target already exists
-            // because it's currently be written to disk in another
-            // request. It's best to just fail silently.
+        } catch (FilesystemV2Exception $exception) {
```

Comment:
Is it normal that we don't handle the edge case anymore?

Reply:
In Flysystem 2 there is no distinction between writing and updating files. Flysystem 1 would fail to write to existing file with FileExistsException, while in Flysystem 2 `write()` works like `put()` in first version - it either creates or updates file. Therefore FileExistsException does not exists anymore and the process can only throw if writing failed, so edge case probably doesn't exists.

But if we want to avoid writing over existing files I can wrap `write` call inside `!fileExists` if clause - I think it can be beneficial.